dbeaver-bin fails to start in master branch

### Describe the bug
dbeaver does not start

### Steps To Reproduce
Steps to reproduce the behavior:
1. run dbeaver

### Expected behavior
it's gui appears

### Additional context
```console
~ via  v3.12.4 via 💠 default 
❯ dbeaver .
WARNING: Using incubator modules: jdk.incubator.vector, jdk.incubator.foreign
Warning: NLS unused message: Workbench_missingPropertyMessage in: org.eclipse.ui.internal.messages
> Start Eclipse Jobs Mechanism [org.eclipse.core.jobs 3.15.300.v20240418-0734]
> Start Equinox Java Authentication and Authorization Service (JAAS) [org.eclipse.equinox.security 1.4.300.v20240419-2334]
> Start DBeaver Application Standalone [org.jkiss.dbeaver.ui.app.standalone 24.1.3.202407211303]
> Start Интерфейс приложения IDE Eclipse [org.eclipse.ui.ide.application 1.5.400.v20240416-0658]
> Start DBeaver UI Editors - Base [org.jkiss.dbeaver.ui.editors.base 1.0.135.202407211303]
2024-07-29 18:49:19.742 - Can't process commands because no running instance is present
SWT OS.java Error: Failed to load swt-pi3, loading swt-pi4 as fallback.
Dbeaver:
An error has occurred. See the log file
/home/spiage/.eclipse/1948949714_linux_gtk_x86_64/configuration/1722268159354.log.
```

```console
~ via  v3.12.4 via 💠 default 
❯ cat /home/spiage/.eclipse/1948949714_linux_gtk_x86_64/configuration/1722268159354.log
!SESSION 2024-07-29 18:49:19.246 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.11
java.vendor=N/A
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=ru_RU
Framework arguments:  . -eclipse.keyring /home/spiage/.local/share/DBeaverData/secure/secure_storage
Command-line arguments:  -os linux -ws gtk -arch x86_64 .

!ENTRY org.eclipse.osgi 4 0 2024-07-29 18:49:19.757
!MESSAGE An error occurred while automatically activating bundle org.eclipse.ui.ide (164).
!STACK 0
org.osgi.framework.BundleException: Exception in org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start() of bundle org.eclipse.ui.ide.
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:854)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
        at org.eclipse.osgi.container.Module.doStart(Module.java:639)
        at org.eclipse.osgi.container.Module.start(Module.java:498)
        at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
        at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
        at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
        at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:375)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:351)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:191)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1454)
Caused by: java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
        no swt-pi4-gtk-4965r11 in java.library.path: /etc/sane-libs
        no swt-pi4-gtk in java.library.path: /etc/sane-libs
        no swt-pi4 in java.library.path: /etc/sane-libs
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk-4965r11.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4.so

        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
        at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
        at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
        ... 31 more
Root exception:
java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
        no swt-pi4-gtk-4965r11 in java.library.path: /etc/sane-libs
        no swt-pi4-gtk in java.library.path: /etc/sane-libs
        no swt-pi4 in java.library.path: /etc/sane-libs
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk-4965r11.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4.so

        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
        at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
        at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
        at org.eclipse.osgi.container.Module.doStart(Module.java:639)
        at org.eclipse.osgi.container.Module.start(Module.java:498)
        at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
        at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
        at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
        at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:375)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:351)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:191)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1454)

!ENTRY org.eclipse.osgi 4 0 2024-07-29 18:49:19.759
!MESSAGE Application error
!STACK 1
java.lang.NoClassDefFoundError: org/eclipse/ui/internal/ide/ChooseWorkspaceData
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:375)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:351)
        at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:191)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:668)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:605)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1481)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1454)
Caused by: java.lang.ClassNotFoundException: An error occurred while automatically activating bundle org.eclipse.ui.ide (164).
        at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:134)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
        at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
        at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
        ... 16 more
Caused by: org.osgi.framework.BundleException: Exception in org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start() of bundle org.eclipse.ui.ide.
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:854)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
        at org.eclipse.osgi.container.Module.doStart(Module.java:639)
        at org.eclipse.osgi.container.Module.start(Module.java:498)
        at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
        at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
        ... 24 more
Caused by: java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
        no swt-pi4-gtk-4965r11 in java.library.path: /etc/sane-libs
        no swt-pi4-gtk in java.library.path: /etc/sane-libs
        no swt-pi4 in java.library.path: /etc/sane-libs
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk-4965r11.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
        Can't load library: /home/spiage/.swt/lib/linux/x86_64/libswt-pi4.so

        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
        at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
        at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
        at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
        at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
        ... 31 more
```

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@gepbird @mkg20001 @yzx9

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
~ via  v3.12.4 via 💠 default 
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.2, NixOS, 24.11 (Vicuna), 24.11.20240729.4d67eb3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `""`
 - nixpkgs: `/nix/store/ril0qzscfs2i45z7akg2ssj7xa3mrzz7-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/db/dbeaver-bin/package.nix
+++ b/pkgs/by-name/db/dbeaver-bin/package.nix
@@ -8,6 +8,9 @@
   gnused,
   autoPatchelfHook,
   wrapGAppsHook3,
+  gtk3,
+  swt,
+  glib,
 }:
 
 stdenvNoCC.mkDerivation (finalAttrs: {
@@ -59,7 +62,15 @@ stdenvNoCC.mkDerivation (finalAttrs: {
         cp -r * $out/opt/dbeaver
         makeWrapper $out/opt/dbeaver/dbeaver $out/bin/dbeaver \
           --prefix PATH : "${openjdk17}/bin" \
-          --set JAVA_HOME "${openjdk17.home}"
+          --set JAVA_HOME "${openjdk17.home}" \
+          --prefix CLASSPATH : "$out/dbeaver/plugins/*:${swt}/jars/swt.jar" \
+          --prefix LD_LIBRARY_PATH : "$out/lib:${
+            lib.makeLibraryPath [
+              swt
+              gtk3
+              glib
+            ]
+          }"
 
         mkdir -p $out/share/icons/hicolor/256x256/apps
         ln -s $out/opt/dbeaver/dbeaver.png $out/share/icons/hicolor/256x256/apps/dbeaver.png

